substrate/frame/revive/src/evm/call.rs

```diff
@@ -42,27 +45,44 @@ pub struct CallInfo<T: Config> {
 	pub tx_fee: BalanceOf<T>,
 	/// The additional storage deposit to be deposited into the txhold.
 	pub storage_deposit: BalanceOf<T>,
-	/// The gas limit as supplied in the transaction.
-	pub gas: U256,
+	/// The ethereum gas limit of the transaction.
+	pub eth_gas_limit: U256,
+}
+
+/// Mode for creating a call from an ethereum transaction.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum CreateCallMode {
+	/// Mode for extrinsic execution. Carries the encoding length of the extrinsic and the
+	/// RLP-encoded Ethereum transaction
+	ExtrinsicExecution(u32, Vec<u8>),
+	/// Mode for dry running
+	DryRun,
 }
 
 /// Decode `tx` into a dispatchable call.
+///
+/// signed_transaction is Some(..) for extrinsic execution (when called from
+/// `try_into_checked_extrinsic`) and it is `None` for dry running (when called from
+/// `dry_run_eth_transact`)
 pub fn create_call<T>(
 	tx: GenericTransaction,
-	signed_transaction: Option<(u32, Vec<u8>)>,
-	apply_weight_cap: bool,
+	mode: CreateCallMode,
```

Comment:
maybe this should be `GenericTransaction::into_call(self, mode: CreateCallMode)`

Reply:
Good, I changed it that way (commit https://github.com/paritytech/polkadot-sdk/pull/10166/commits/0f55b11e95e1d709402c8738cbae688060e7a8ef).